TESTING.md

```diff
@@ -0,0 +1,50 @@
+* Where builtin functions have been mocked, be sure to clear them between tests.
+* Test at an appropriate level of isolation. Eg. 
+    * If you’re testing a view, you do not need to test the whole application UI, you can just fetch the view provider using the public API and render the view into an element that you have created. 
+    * You do not need to test that the view switcher works, there should be separate tests for that. 
+    * You do not need to test that telemetry providers work, you can mock openmct.telemetry.request() to feed test data to the view.
+    * Use your best judgement when deciding on appropriate scope.
+* Automated tests for plugins should start by actually installing the plugin being tested, and then test that installing the plugin adds the desired features and behavior to Open MCT, observing the above rules.
+* All variables used in a test spec, including any instances of the Open MCT API should be declared inside of an appropriate block scope (not at the root level of the source file), and should be initialized in the relevant beforeEach block. `beforeEach` is preferable to `beforeAll` to avoid leaking of state between tests.
+* A `afterEach` or `afterAll` should be used to do any clean up necessary to prevent leakage of state between test specs. This can happen when functions on `window` are wrapped, or when the URL is changed. [A convenience function](https://github.com/nasa/openmct/blob/master/src/utils/testing.js#L59) is provided for resetting the URL and clearing builtin spies between tests.
+
+#### Unit Test Examples
+* [Example of an automated test spec for an object view plugin](https://github.com/nasa/openmct/blob/master/src/plugins/telemetryTable/pluginSpec.js)
+* [Example of an automated test spec for API](https://github.com/nasa/openmct/blob/master/src/api/time/TimeAPISpec.js)
+
+#### Unit Testing Execution
+
+The unit tests can be executed in one of two ways:
+`npm run test` which runs the entire suite against headless chrome
+`npm run test:debug` for debugging the tests in realtime in an active chrome session.
+
+## e2e, performance, and visual testing
+Documentation located [here](./e2e/README.md)
+
+## Code Coverage
+
+* 100% statement coverage is achievable and desirable.
+
+Codecov.io will combine each of the above commands with [Codecov.io Flags](https://docs.codecov.com/docs/flags). Effectively, this allows us to combine multiple reports which are run at various stages of our CI Pipeline or run as part of a parallel process.
+
+This e2e coverage is combined with our unit test report to give a comprehensive (if flawed) view of line coverage.
+
+### Limitations in our code coverage reporting
+
+Our code coverage implementation has two known limitations:
+- [Variability and accuracy](https://github.com/nasa/openmct/issues/5811)
+- []
```

Comment:
This seems to be incomplete?

Reply:
@akhenry documentation is incomplete but the rest is ready for review